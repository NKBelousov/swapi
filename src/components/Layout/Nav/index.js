import React, { Component } from "react";
import { map } from "lodash";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const Navigation = styled.nav`
  background: red;
  color: #fff;
  font-size: 2em;
  padding: 1em;
`;

const Link = styled.a`
  border-bottom: 1px solid #fff;
  cursor: pointer;
  margin: 0 0.5em;
  text-align: center;
`;

class Nav extends Component {
  static get propTypes() {
    return {
      items: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          url: PropTypes.string.isRequired,
        })
      ).isRequired,
    };
  }
  constructor(props) {
    super(props);
  }
  onLink(url) {
    window.location = url;
  }
  render() {
    return (
      <Navigation>
        {map(this.props.items, l => {
          return (
            <Link key={l.url} onClick={this.onLink.bind(this, l.url)}>
              {l.name}
            </Link>
          );
        })}
      </Navigation>
    );
  }
}

export default Nav;
