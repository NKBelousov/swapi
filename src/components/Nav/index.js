import React, { Component } from "react";
import { map } from "lodash";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const Navigation = styled.nav`
  background: red;
  color: #fff;
  font-size: 2em;
  padding: 1em;
`;

const Link = styled.a`
  border-bottom: 1px solid #fff;
  cursor: pointer;
  margin: 0 0.5em;
  text-align: center;
`;

class Nav extends Component {
  static get defaultProps() {
    return {
      links: [
        {
          name: "Home",
          url: "/home",
        },
        {
          name: "About",
          url: "/about",
        },
        {
          name: "Contacts",
          url: "/contacts",
        },
      ],
    };
  }
  constructor(props) {
    super(props);
  }
  onLink(url) {
    window.location = url;
  }
  render() {
    return (
      <Navigation>
        {map(this.props.links, l => {
          return (
            <Link key={l.url} onClick={this.onLink.bind(this, l.url)}>
              {l.name}
            </Link>
          );
        })}
      </Navigation>
    );
  }
}

export default Nav;
