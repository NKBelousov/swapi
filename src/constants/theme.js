export const PRIMARY = "#db2c2c";
export const SECONDARY = "#fff";

export default Object.freeze({
  primary: PRIMARY,
  secondary: SECONDARY,
});
