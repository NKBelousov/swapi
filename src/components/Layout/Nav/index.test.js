import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";

import theme from "~/constants/theme";
import Nav from "./index.js";

test("Should render correctly", () => {
  const ITEMS = [
    {
      url: "/first",
      name: "First",
    },
    {
      url: "/second",
      name: "Second",
    },
  ];
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>
          <Nav
            items={ITEMS}
            onPeople={alert}
            onPlanets={alert}
            onFilms={alert}
          />
        </Router>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
