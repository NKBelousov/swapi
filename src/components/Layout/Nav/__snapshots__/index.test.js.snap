// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<nav
  className="sc-bdVaJa bPufhP"
>
  <a
    className="sc-bwzfXH jCgOeb"
    href="/first"
    onClick={[Function]}
  >
    First
  </a>
  <a
    className="sc-bwzfXH jCgOeb"
    href="/second"
    onClick={[Function]}
  >
    Second
  </a>
</nav>
`;
